framework/src/controller/channels/in_memory_channel.ts

```diff
@@ -108,6 +108,10 @@ export class InMemoryChannel extends BaseChannel {
 					const stateStore = new PrefixedStateReadWriter(this._db.newReadWriter());
 					return stateStore.getStore(moduleID, storePrefix);
 				},
+				getOffchainStore: (moduleID: Buffer, storePrefix?: number) => {
+					const stateStore = new PrefixedStateReadWriter(this._db.newReadWriter());
```

Comment:
`this_db` here refers to the `StateDB` but we need `moduleDB` passed to this class and used for the `offchain store`.
Also, if we do so, `PrefixedStateReadWriter` will not make sense so you can use the `StateStore` from the `lisk-chain`

Reply:
How about `getStore`? Should remain as-is right? `this._db` refers to `StateDB`?

Reply:
yes, `getStore` should return the StateDB as it is. just offchainStore need to use the module id

Reply:
Sorry was in the process of updating the comment 😃...thanks for the response